### DISCUSSION TITLE
Starlink install tips

### DISCUSSION THREAD

**Discussion by Chris5342** [Ysurac/openmptcprouter#2353] [Show and tell]
Jan 5, 2023 : edited to simplify.

It took a while to get Starlink going. Hopefully this will help others.
(It is not intended to be a full tutorial in setting up OpenMPTCProuter)

This was done with Starlink V2 (rectangular dish) but should also be applicable to V1 (round dishes).
I use a Raspberry Pi but it should work with other devices.

**OpenMPTCProuter’s main address**

Dishy and the app uses 192.168.100.1 (and it can’t be changed) so first change OpenMPTCProuter’s main address :

Network -> Interfaces -> Lan -> Edit
![Screenshot 2022-05-17 at 13 57 59](https://user-images.githubusercontent.com/105393840/168825938-4b0a267c-1831-4d64-849a-c5c982e75901.jpg)

Use 192.168.*.1 where * is a number not used elsewhere on your LAN. I used 42 because another user did.
Press Save and reboot the Pi and make sure you can connect.

**Connecting the Starlink to the LAN**

Unfortunately you cannot turn off the Starlink DHCP router so it is best to use a dedicated LAN port, in my case this meant with a USB to LAN port adapter.  

You can use a USB port on the Pi (eg, TP-Link UE300) :
![Screenshot 2022-05-17 at 14 18 28 copy](https://user-images.githubusercontent.com/105393840/169825941-17558590-2e6f-4fd8-82ca-32f336cc0015.jpg)

The Ethernet/USB settings are (in the Wizard - System -> OpenMPTCProuter )
![Screenshot 2022-05-23 at 14 09 09](https://user-images.githubusercontent.com/105393840/169826558-8ba9f81d-07c0-47e3-98c4-5401270eca11.jpg)

Hopefully Starlink will now work!

**Starlink app**

To use the starlink app you need to setup a static route for 192.168.100.1 :

On OpenMPTCProuter, Network -> Static routes -> Static IPv4 routes :
![Screenshot 2022-05-17 at 14 38 05](https://user-images.githubusercontent.com/105393840/168826552-867554c2-60d2-49fd-8361-19688092ce5a.jpg)

If anyone can help enhance this please let me know (post below) and I will update.

Hope this will help someone.

**Comment by robroy90** [comment]
Thanks! This is good info! I appreciate the details! I don't use the Starlink power supply/WiFi box. I use PoE with Dishy instead.

**Reply by Chris5342** [reply]
What's the reason for not using the Starlink power supply?

**Reply by darkmaniac7** [reply]
> What's the reason for not using the Starlink power supply?

I do this as well. The lead time on an Ethernet Adapter from starlink was several weeks, I also didn't want another Point of Failure with the Router that you can't really access or change anything on. I also wasn't sure on the surge supression of the router when supplying PoE. 


I just have a Mean Well HLG-120H-48A as the Power Supply and a 800-GIGE-POE-APC Injector.

**Comment by Kirball** [comment]
Are you not getting bottlenecks using a raspberry pi.  I bought a banana pi r-64 because there is limitations to rpi and it's lan speeds (at least in the version 3 that I've used in the past) unfortunately the firmware for the b pi doesn't work and just gets stuck in a boot loop and won't run at all.

**Reply by Chris5342** [reply]
You don't say how fast you are trying to run.
I have Starlink, 4G (0-20M) and ADSL (3M) so get about 50-200Mbit/s and the Pi4B seems fine.

I was seeing Bufferbloat (when Starlink goes wobbly) but it looks like SQM as advised by @Ysurac here: https://github.com/Ysurac/openmptcprouter/discussions/2357#discussioncomment-2833963 has helped.

I am still testing  this and plan to do a show and tell when I have a better handle on it. 

Not sure this will help.

**Reply by Kirball** [reply]
I believe the LAN ports on older models have a max speed of 100 mbps.  I need as much upload speed as possible for twitch streaming and my home server with multiple game servers.  Of course I'll never get 100 mbps upload with starlink and my 4G LTE home router but I get over 100 mbps down from starlink from time to time.

**Comment by ioogithub** [comment]
> You can use a USB port on the Pi (eg, TP-Link UE300) : ![Screenshot 2022-05-17 at 14 18 28 copy](https://user-images.githubusercontent.com/105393840/169825941-17558590-2e6f-4fd8-82ca-32f336cc0015.jpg)

Thanks @Chris5342 for this guide. I am considering getting starlink to use with openmptcprouter. I was considering purchasing one of the arm based boards that openmptcprouter supports such as the expressobin (https://github.com/Ysurac/openmptcprouter/discussions/2496) but it sounds like the pi is the only one with a strong enough processor to handle the packet routing.

My question is about using a USB hub such as the TP-Link you mentioned. Would this then become a bottleneck? USB 3.0 has a max bandwidth of 6Gb which I believe the pi shares with all ports would latency or lack of bandwidth become a bottleneck?

How have your results been with startlink and using the pi? Did you bond it with a 4G and a DSL as per your diagrams? I would be purchasing the startlink hardware, pi and USB adapter for this setup, would you recommend it?

**Reply by Chris5342** [reply]
My speed varies quite a bit. But my mesh WIFI is a bit iffy and I also have ethernet over mains in the loop. I sometimes get 200M and typically 50-100M. I do not think the PI etc is a bottleneck. Sorry if that's a poor answer but 50M is actually ok for most of my needs.

I do bond with 4G and DSL. The 4G has been very flaky with lots of high pressure and I usually set it to backup as Starlink is pretty good. I'm not sure I would add 4G if I did not already have it.

What I can say (and am planning a show and tell when I get a chance) is "redundant mode". Redundant mode uses all links in parallel sending copies of packets down all links. I have found this is useful for video conference. I think it fixes any glitches in Starlink.

This is the setup:

/cgi-bin/luci/admin/network/mptcp
![Screenshot 2022-09-01 at 22 29 28](https://user-images.githubusercontent.com/105393840/188016539-9c7c49f1-c03d-403f-8d64-7323a8e997d2.jpg)

Pretty simple and doesn't seem to be much known about.

**Reply by ioogithub** [reply]
Thanks for the reply!  I look forward to seeing your redundant setup. I ordered start link "best effort" today and plan to bond it with a reliable but slower existing 4G 20/6 Mbps line using OMP.

For the hardware, I was initially thinking about the expressobin or banannapi as they have multiple RJ45 ports but however I think the raspberry pi 4 is a better choice as it has a stronger processor for packet routing as per @Ysurac's advice.

I am planning on using use two TP-Link UE300 adapters as you mentioned. I believe the kernel supports the Realtek 8153 chip but did you have to install a driver or did the openmptcprouter image support it out of the box?

**Reply by Chris5342** [reply]
Plug and play :)

Note I got the UE300 tip here: https://wirelessjoint.com/viewtopic.php?p=24681&sid=184e08a25aabd27023b72cc6eacb3eba#p24681

Probably worth a read.

**Reply by ioogithub** [reply]
> Plug and play :)
> 
> Note I got the UE300 tip here: https://wirelessjoint.com/viewtopic.php?p=24681&sid=184e08a25aabd27023b72cc6eacb3eba#p24681
> 
> Probably worth a read.

Excellent thread, thanks for posting this, I was unaware of this site. Although I love the idea of using a pi, like the posters in that thread mentioned, I am also starting to consider a beefier x86 mini PC like a Protectli or similar because now I am doubting the potential bandwidth of a pi with USB adapters. Also the price of pi's has tripled over the last year :(

I found this issue: https://github.com/Ysurac/openmptcprouter/issues/2016 where @gsxryan posted huge performance decrease when using a pi with USB adapters:

from:
single port macVLANs:  no encryption: 600/690
to:
multiport (add: usb3.0 eth1: no encryption 100 / 5

If I am understanding his test result correctly, in the first test he was using the single Pi's Ethernet port with VLANS and in the second test he used a physical USB adapter. The results don't look great, I wonder what the huge bottleneck is with the adapter. @gsxryan if you are out there, did you ever repeat the test, was there a hardware or driver issue or is this an expected result when using these adapters?

On the other hand you mentioned that sometimes you get 200 with those adapters so that is promising.

**Reply by adamb** [reply]
> https://wirelessjoint.com/viewtopic.php?p=24681&sid=184e08a25aabd27023b72cc6eacb3eba#p24681

Any tips on adding a new interface?  I bought the E300 and plug it into the Pi.  But I'm stuck at that point. There is no new interfaces showing up.  I assume I have to add them somehow, but don't know what to do.  Any pointers would be appreciated!  (I've spent over an hour googling and looking and watching some videos but still don't get it..  I'm a programmer but not really much of a sys admin.)

**Reply by iambenmitchell** [reply]
> Any tips on adding a new interface?  I bought the E300 and plug it into the Pi.



Can you remind me what the E300 is? Is it a smart switch? 

How are you connecting these interfaces to the pi? Are you using VLANs or USB?

**Reply by Chris5342** [reply]
https://www.amazon.com/TP-Link-Foldable-Gigabit-Ethernet-Compatible/dp/B00YUU3KC6

Plugged into USB 3

Just worked for me.

**Reply by iambenmitchell** [reply]
> https://www.amazon.com/TP-Link-Foldable-Gigabit-Ethernet-Compatible/dp/B00YUU3KC6
> 
> 
> 
> Plugged into USB 3
> 
> 
> 
> Just worked for me.
> 
> 

Don't forget to disable interface renaming if you do this

**Reply by adamb** [reply]
> > https://www.amazon.com/TP-Link-Foldable-Gigabit-Ethernet-Compatible/dp/B00YUU3KC6
> > Plugged into USB 3
> > Just worked for me.
> 
> Don't forget to disable interface renaming if you do this

I see it as eth1 or eth2 or sometimes eth3.  Is this what you mean by disable interface renaming?  Can you tell me how to do that?  I googled 'disable interface renaming' and didn't find anything relevant.  

I have two E300s.  One for the LTE and one for Starlink....  I seem to be able to get them connected but the interface (hardware?) names seem to change.

**Reply by iambenmitchell** [reply]
> I see it as eth1 or eth2 or sometimes eth3.  Is this what you mean by disable interface renaming?  Can you tell me how to do that?  I googled 'disable interface renaming' and didn't find anything relevant.  

Check in System->OpenMPTCProuter, "Advanced settings" tab "Disable interfaces auto rename"

If it's ticked, untick it. If it is already unticked, then tick it as I can't remember if you want it to be on or off. Just invert whatever it is set to at the moment and then reboot. Should solve your issues I think

**Reply by iambenmitchell** [reply]
I might have missunderstood your question. If you're just trying to add a new WAN connection then press this in System->OpenMPTCProuter, "setup wizard" 

Select the new interface from the drop down then press "add an interface". Fill out the details it needs (presumably DHCP?) and then press save&apply

![image](https://user-images.githubusercontent.com/27426792/202872614-e0c81bf9-0325-4895-97b7-64ebfd7bd68c.png)

**Comment by gsxryan** [comment]
Hi @ioogithub.  I only repeated the test a few times during that session to confirm.  But have not rebuilt that scenario since then.  

It seems the USB controller on the RPIs is just not good enough to handle the speeds.  This is why i tested to see if VLANs on the single port would be better. 

Jeff Geerling did some really good research on this.  And i believe some of his videos explain.  Although i cant remember the exact video, Check out his article here; https://www.jeffgeerling.com/blogs/jeff-geerling/getting-gigabit-networking

**Comment by gsxryan** [comment]
Also, on topic - i'm using Starlink on square Dishy V2.  I took the approach of rather changing the IP address of OMR to 192.168.101.1 - Leaving Starlink config completely alone.  I run Ethernet to my OMR switch, and also still run the Starlink Wifi as an alternative backup.

**Reply by ioogithub** [reply]
> Also, on topic - i'm using Starlink on square Dishy V2. I took the approach of rather changing the IP address of OMR to 192.168.101.1 - Leaving Starlink config completely alone. I run Ethernet to my OMR switch, and also still run the Starlink Wifi as an alternative backup.

This is the approach I was considering as well although I haven't considered running the starlink wifi, is it possible to deactivate this? 

Also there is discussion of using the Startlink Ethernet adapter. I am unclear as to what this is for exactly. Do I need to order this from Starlink if I am planning on using OMR?

**Reply by gsxryan** [reply]
Wifi can be deactivated in the starlink app once you connect the Ethernet adapter.

Dishy V2 has a proprietary connector required to get wired.  Here's a picture;  But please don't pay these scalpers.  It's much cheaper on starlink's store, but its locked behind your account access.
https://www.amazon.com/Starlink-Ethernet-Adapter-External-Network/dp/B0B4KSBR7L

**Reply by ioogithub** [reply]
> Dishy V2 has a proprietary connector required to get wired. Here's a picture

Hmm so do I need to purchase this separately if I want to use starlink and bond it with another line with openmptcprouter? I just looked at the confirmation order email and there is a line drawing of all the parts that will be shipped. There are two cables, one looks like a power cable and one looks like an Ethernet cable but the Ethernet cable does not have the big box on one end like the amazon photo, is this something different?

**Comment by ioogithub** [comment]
Hi @Chris5342

Got my dishy today and dove right in. I am following your guide but am stuck.

So far:
x Startlink is online, in bypass mode, internet access works.
x OMR router image booted on raspberry pi 4
x two TP-Link UE300 connected to the two USB3 ports on the pi

I am stuck on this part:
```
If using a Ethernet/USB settings are:
Screenshot 2022-05-23 at 14 09 09
```

What is the menu location for this screen? I thought I might have to Network->Interfaces and create an an interface with eth1 as the device for the dongle starlink is attached to but the menu looks substantially different than your screenshot. Am I in the right place?

Also, if I do a 'ip address' at the command prompt I do see eth1 and eth2 but I am not certain they correspond to the dongles or they are something else, they are down anyway. Did you have to install this package to get them to work?  https://openwrt.org/packages/pkgdata/kmod-usb-net-rtl8152

**Reply by GrantSparks** [reply]
There is a common chipset that many USB dongles use which seems to have a compatibility problem with RPi4 and OpenWRT.  The workaround to get the interfaces to come up is to not use the USB3 ports but plug them into the USB2.0 ports instead.   

Not sure if that's part of your issues but maybe it will help.

**Reply by Ysurac** [reply]
Screenshot come from System->OpenMPTCProute, wizard tab.

**Reply by ioogithub** [reply]
> Screenshot come from System->OpenMPTCProute, wizard tab.

Thank you I got it working! No drivers needed. This is the most exciting project I have worked on this year, that VPS install script was a work of art, very exciting watching it install!

Everything seems to be working except the static route of the starlink statistics page :

> On OpenMPTCProuter, Network -> Static routes -> Static IPv4 routes : ![Screenshot 2022-05-17 at 14 38 05](https://user-images.githubusercontent.com/105393840/168826552-867554c2-60d2-49fd-8361-19688092ce5a.jpg)

I have setup a static route but no matter what I do I can't access this page @ 192.1698.100.1 when the two connections are bonded (I changed the OMR server IP to 192.168.200.1 as the starlink admin page can't be changed). The page loads when the PC is connected directly to starlink. This page is pretty critical because I installed OMR right away so I have no idea how dishy is actually doing, if I have obstructions etc.

@Chris5342, is this screenshot showing how to foward the page with USB adapters or with a router in between starlink and OMR?

I have this topology:

`starlink -> OMR (rpi4 via USB via UE300) -> Mesh network (OpenWRT on a GL-B1300) -> PC`

and this static route:
```
Interface	Disable	Target	IPv4-Netmask	IPv4-Gateway	Metric	On-Link route	
lan: ["eth0"] 	No	192.168.100.1	255.255.255.0	192.168.200.1	0	No`
```
I can't ping 192.168.100.1 from  any hop (OMR, GL-B1300 or the PC).  I see several users have got this working, does anyone have any other tips? Is my gateway wrong I tried several different ones.

**Reply by ioogithub** [reply]
> Everything seems to be working except the static route of the starlink statistics page :

So I found the solution myself. The gateway in the screenshot has to be the gateway of the starlink device. So from this page: https://192.168.200.1/cgi-bin/luci/admin/system/openmptcprouter/status look at the starlink line, it is the IP address at the top in bold. In my case it started with 100.64.

The other issue is that I had a route stuck in the routing table (root@OpenMPTCProuter:~# route -n) from trying too many attempts to add a static route. After I rebooted it worked.

**Comment by ioogithub** [comment]
Running some benchmark speed tests it looks like I have a bottleneck. OMR's aggregated bandwidth is less than the single starlink alone:

`
omr-test-speed
`
- 2 min run time
- ping < 50ms from VPS:

### Results:
eth1 (starlink):  **14.0M**
eth2 (LTE) **2872k**
OMR **7648k**

eth1 (starlink): **11.9M**
eth2 (LTE) **2943k**
OMR **8835k**

@Ysurac: What would you guess the bottle neck is caused by with this setup:

- OMR server = raspberry pi 4b 8gb
- eth1 = starlink via TP-UE300 USB 3.0
- eth2 = 4G via TP-UE300 USB 3.0
- VPS = linode single core shared

### Bottleneck Guesses:
1. RPI processor not being able to encrypt/decrypt and route packets fast enough?
- Running `top` on OMR during speed test shows very low < 1-2% CPU
- load: 1.44 0.98 0.65
-  pi is running  cool at 47c without active cooling 

2. USB 3.0 bandwidth constraints on the TP-UE300 dongles
- I think it is 6GiB shared total bandwidth on this bus?

3. VPS processor or bandwidth constraints
- Increase to 2 or 4 cores?

I still need to purchase a permanent solution so either another rpi or x86.

Two devices that look made for this application, similar price and both have aes-ni support:
- Protectli Vault FW4B (Celeron J3160)
- QOTOM Q330G4 (Core i3)

Would the stronger i3 processor be worth it? I would be interested to hear your thoughts on what I should try next.

**Reply by Ysurac** [reply]
RPI4 can go up to 300-350Mb/s. So it's not the bottleneck here.
I don't like using USB on RPI, it's often a problem.
For the VPS one fast core is better than 2 slow cores. Check speed on VPS using "omr-test-speed" on it, check also cpu usage. It's difficult to verify if VPS CPU is fast enough when CPU are shared (and long time since I tested Linode, no idea of speed on shared VPS...).

What is the result if you disable LTE interface ?

For device, a x86_64 CPU with AES-NI is always better than a RPI (and for now can be cheaper...). Better to have fast core CPU than multiples slow cores.

**Comment by ioogithub** [comment]
> Check speed on VPS using "omr-test-speed" on it, check also cpu usage.

## VPS Test
Bandwidth doesn't seem to be a problem on the VPS:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9536M  100 9536M    0     0   569M      0  0:00:16  0:00:16 --:--:--  639M
```

During the test `top` showed the curl process was consuming around 15-20% on average with a 1-2 second peak to 30% with loads:
0.23 0.11 0.3

> What is the result if you disable LTE interface ?
## Speed test with LTE disabled
- Network->Interface->Stop

Results:

### 1
`omr-test-speed`
  ```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  1 9536M    1  145M    0     0  1165k      0  2:19:35  0:02:07  2:17:28 3860k^C
```
`omr-test-speed eth1`
  ```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  222M    0  222M    0     0  1605k      0 --:--:--  0:02:21 --:--:--  335k^C
```
### 2
`omr-test-speed`
  ```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  328M    0  328M    0     0  2629k      0 --:--:--  0:02:07 --:--:-- 3798k^C
```
`omr-test-speed eth1`
  ```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  5 9536M    5  498M    0     0  4032k      0  0:40:21  0:02:06  0:38:15 3057k^C
```
### 3
`omr-test-speed`
  ```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  3 9536M    3  331M    0     0  1313k      0  2:03:54  0:04:18  1:59:36  586k^C
```
`omr-test-speed eth1`
```
Best server is http://speedtest.newark.linode.com/garbage.php?ckSize=10000, running test:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  307M    0  307M    0     0  2549k      0 --:--:--  0:02:03 --:--:-- 6503k^C
```
### 4
`omr-test-speed eth`
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  2 9536M    2  278M    0     0  2348k      0  1:09:17  0:02:01  1:07:16 3598k^C
```
`omr-test-speed eth1`
 ```
 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 20 9536M   20 1990M    0     0  15.9M      0  0:09:58  0:02:05  0:07:53 12.9M^C
```

### 5
`omr-test-speed eth1`
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  698M    0  698M    0     0  3853k      0 --:--:--  0:03:05 --:--:-- 10.4M^C
```
`omr-test-speed
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  295M    0  295M    0     0  2282k      0 --:--:--  0:02:12 --:--:-- 1898k^C
```

Starlink is all over the place but in each test except #5 `omr-test-speed` was slower than `omr-test-speed eth1` Also, eth2 (disabled during this test) is the Multipath TCP Master, not sure if this affects the test.

During these tests on OMR, `top` on the VPS showed very little CPU, various OMR processes taking 1-2% loads: 0.07 0.4 0.5 so CPU on VPS doesn't appear to be the bottleneck.

**Comment by ioogithub** [comment]
I found an two article where the tester adjusted IRQ and CPU scheduling on the pi and ran routing and network performance tests with a USB dongle that uses the same chip as the one mentioned in this thread. His conclusion is that the pie is capable of gigabit routing performance:
https://tomcore.io/docs/articles/RaspberryPI/raspberry-pi-4b-routing-performance/
https://tomcore.io/docs/articles/RaspberryPI/raspberry-pi-4b-usb-network-performance/

Did you have to adjust these setting to get the 300-350 Mb/s you mentioned?

I am only dealing with around an average of 50 Mb/s. I have been testing all day but OMR is always giving me speeds slower than the starlink alone. I also tested stability, there are no packet drops and latency is similar across both connections (40-80ms on both devices)

@Ysurac Do the results of my speedtest indicate any potential issues or other things I can try?

**Reply by Ysurac** [reply]
To get 350Mb/s, default OpenMPTCProuter configuration work. The only change I have is using BBR2 as TCP Congestion Control in Network->MPTCP.
Did you try with another VPS provider ?

**Reply by ioogithub** [reply]
> To get 350Mb/s, default OpenMPTCProuter configuration work. The only change I have is using BBR2 as TCP Congestion Control in Network->MPTCP.

I will try this and see if it makes any difference.

> Did you try with another VPS provider ?

No but I can try that, do you think that is the issue? From the benchmark I posted above, the VPS provider is giving me average bandwidth of 569M, less than 50ms latency while I am only seeing 30M-50M average speeds. Do you think the VPS is the bottleneck considering this data?

If it is not the VPS provider, what else can cause the aggregate bandwidth to be slower than the fastest single wan connection?

**Reply by b4cc** [reply]
Hello, I'm getting the same behavior with Starlink, my bandwidth with OMR is about 100Mbits and without OMR is about 250Mbits. I tried with several VPS, did you find the issue?

**Reply by iambenmitchell** [reply]
> Hello, I'm getting the same behavior with Starlink, my bandwidth with OMR is about 100Mbits and without OMR is about 250Mbits. I tried with several VPS, did you find the issue?


What is the speed of the other connections? 

Your speed is limited by the slowest connection. 

If you have starlink doing 250mbps and combine it with a 4G dongle doing 50mbps then you'll only be able to get 50mbps from starlink (100mbps total)

**Reply by b4cc** [reply]
My configuration is:
ADSL2+ (~6Mbits Download speed)
VDSL2 (~17Mbits Download speed)
Starlink (~250Mbits Download speed)

I saw that, disabling the encryption, the speed has increased to 150Mbit.

**Reply by iambenmitchell** [reply]
OMR isn't for you because those speeds aren't remotely similar.

You're best off using pfsense or something similar and doing policy routing where you use your Starlink as your Main WAN and the others as a failover or for lower latency services such as a Gaming PC, Xbox, VOIP phones, Discord etc.

**Reply by Chris5342** [reply]
I have Starlink (50-150M) and 4G/LTE (2-20M) bonded and it seems to work fine.
The 4G is more for a backup than a speed boost.
Its also to increase reliability for video calls using redundant mode.

**Reply by b4cc** [reply]
My target was exactly to increase reliability

**Reply by b4cc** [reply]
I tried also to disable all the WAN less Starlink, but I’m still capped to about 100Mbits.
To be sure that the problem is not my NanoPi R2S, I also tried to go in direct output with just the Starlink, and it worked fine.

@Chris5342, did you change something particular in your OMR configuration from the default settings?

**Reply by Chris5342** [reply]
I think my settings are pretty standard.

A limit of 100Mbits sounds like you are using a 100Mbit port or switch or
there is a driver issue with the NanoPi R2S

Hmm, found this comment online:
_"I have it to use as a router with squid cache, but the Gbit NIC only connect at 100Mbit or less. I can speedtest my network at 220Mbit, then put the R2S in line between my cable modem and router and I got sub 100 speeds, I have tried with FriendlyWRT and DietPi OS loads. I guess I'll relegate it to my guest network so that it's not a total waste"_

**Reply by b4cc** [reply]
<img width="239" alt="image" src="https://user-images.githubusercontent.com/4189231/231089012-f250b4a3-2aad-42e9-8ab2-eea84f694656.png">

 Those changes to the config solved my problems! Now I can reach the max bandwidth.

**Comment by Kirball** [comment]
Personally I'm looking to increase my upload for my streams & for my
servers.  Starlink has no CGNAT so it's a nightmare trying to game with my
friends or set up my content backups if I'm not on my home network.

**Reply by ioogithub** [reply]
@Kirball:

What has been your experience bonding starlink with OMR? Can you achieve a sustained speed higher than the slowest connection? Are you using SQM for bufferbloat, if so what do you set the Upload and Download speeds at?

I see above you mentioned you bought a bananna pi. I was considering this device but ended up with an x86 device with dedicated NICs and a strong CPU based on the recommendations. This should have eliminated several potential bottlenecks but honestly I am seeing the same bandwidth constraints with the new hardware.

I haven't found any settings yet that have really made any substantial difference. When I bond starlink with a 4G connection I always end up with a speed slower than the 4G by itself. The Starlink connection actually seems to destabilize the solid 4G connection when they are bonded as well so OMR isn't improving stability either.

**Comment by ioogithub** [comment]
@Chris5342 Do you have any update on using SQM with starlink and problems with bufferbloat? What values did you set the Download and Upload speeds at? Do you use SQM for all connections or just for Starlink?

**Reply by Chris5342** [reply]
My 4G has got so bad I have set it to backup so not sure my setup is much use. 
As Starlink works pretty well I have not had much incentive to improve it.
These are my settings:

ADSL:
![Screenshot 2022-10-01 at 07 46 40](https://user-images.githubusercontent.com/105393840/193396758-057becc4-4ee4-46b3-96f7-33b6b2920f8c.jpg)

4G:
![Screenshot 2022-10-01 at 07 46 51](https://user-images.githubusercontent.com/105393840/193396768-3f1190b9-c9e5-41b8-949c-3d4cfe27303f.jpg)

SQM Not used for Starlink.

I have not noticed buffer bloat (if that is what it was) for a while but I tend to use redundant mode ( https://github.com/Ysurac/openmptcprouter/discussions/2588 ) for video calls.

**Comment by JeedHome44** [comment]
Hello, 
We can't connect ethernet starlink in OMR LAN ?
With the By Pass mode 192.168.100.1 to Starlink is not disable ?

I have lot of IP in 192.168.100.X with Jeedom and Home Assistant !
I see we have to change OMR IP ! It's a joke ?

But can I leave all my devices at 192.168.100.X and only change the OMR IP?
Will it work to change only the OMR IP?

**Comment by RisaKirisu** [comment]
An update/tip for people who is a newbie like me, has a Starlink Mini and tries to set this up in the latest version (6.12.15) : I completed the same setup with a Starlink Mini. To make the Starlink app work, the static IPv4 route needs to be like this rather than what OP showed in their screenshot:

**Router type: `unicast`
Interface: `wan1` (or whatever interface your Starlink is connected on)
Target: `192.168.100.0/24`
Gateway: `192.168.1.1` (Starlink mini router's IP)**

I don't know if this is the same for other Starlink variants.

**Comment by h45h3ncryp73d** [comment]
My 2 cents ... If you're using Starlink move (run?) away from RPi (at least 4b but my good guess is 5 is not far away but for its price I'd rather go for some i3 NUC) as it will bottleneck it ...

**Reply by RisaKirisu** [reply]
I'm on RPi5, and for purely an router, Starlink's bandwidth (~300Mbps) is far from saturating RPi5's cpu. However, this is not to say that RPi5 is a good choice for the sole purpose of being a router (it's not), just that it wouldn't be a bottleneck.

**Reply by h45h3ncryp73d** [reply]
The point of my comment was not to go into RPi 4b/5 pro/contra discussion. Can it run OMR just fine - yes it can. Furthermore, no doubt RPi represents hardware leap from RPi 4b but it still is what it is ... simply (still) not comparable to x86 platform. Not to go into too many details my advice was a first hand one but unfortunately you've based your comment upon incorrect premise (which we even would share if I haven't tried x86)- Starlink has increased its speed up to 500+ Mbps which cannot be "achieved" on RPi platform, not even RPi5. Try it, you might be surprised by Starlink as I was.
Long story short - to "enjoy" increased Starlink bandwidth you'll need more than RPi.

**Comment by kilazin** [comment]
I have a starlink v3 and mini I am trying to mesh. My situation is i get it going fine. However, once i set the VPS i cannot access the internet at all and my ssh from my VPS locks down and I cannot access the VPS to the internet as well either. To have more context to my problem here is my setup:

Starlink v3 I have in Bypass Mode connected to my OMR(router) 
The OMR Router is home build x86/64 intel i3 with 32 gigs of ram ddr4 i have a dual 10G network card 2 2.5G network card and 1 1G card. 
The OMR Router is then connected to my Cisco Switches 48 POE port switch with 4 10G spf+ which then connects to my other 10G switch 

So I have OMR Router runing fine I had to change the IP for various reasons but mostly cuz i like my home server with my own lan ip.

The OMR Router IP is set to :(for instance) 10.10.1.1 
Starlink routers are in bypass mode so they get no dhcp

I can connect to the internet perfectly aslong as the VPS is not using the shadowsock or shadowsocks rust20111 proxy. I have tried everything regarding that and was wondering what did you do to unblock the ports on shadowsocks proxy. When i use every other proxy or disconect everything i get this shown:

<img width="3252" height="1698" alt="image" src="https://github.com/user-attachments/assets/24280bad-8570-455e-817e-f8b5155f8d58" />

So my guess is a firewall issue with shadowsocks

Any pointers given would be greatly appreciated!

**Comment by ellielo** [comment]
OpenMPTCProuter is a smart way to combine multiple internet connections into one, improving speed and reliability using Multipath TCP. I often read [archeonis.com](https://archeonis.com/) when exploring tech projects, and tools like this really show the power of open-source collaboration. Setting it up isn’t plug-and-play — you need some networking knowledge to handle configuration, latency, and MTU issues. For users with slow or unstable internet, though, it can be a game-changer. It’s a DIY solution that rewards patience and careful setup.
